src/createServer.js

```diff
@@ -1,11 +1,198 @@
 'use strict';
 
-// const express = require('express');
+const express = require('express');
 
 function createServer() {
-  // Use express to create a server
-  // Add a routes to the server
-  // Return the server (express app)
+  const app = express();
+  let users = [];
+  let expenses = [];
+  let nextUserId = 1;
+  let nextExpenceId = 1;
+
+  app.post('/users', express.json(), (req, res) => {
+    const { name } = req.body;
+
+    if (!name) {
+      res.sendStatus(400);
+
+      return;
+    }
+
+    const newUser = {
+      id: nextUserId++,
+      name,
+    };
+
+    users.push(newUser);
+
+    res.status(201);
+    res.send(newUser);
+  });
+
+  app.get('/users', (req, res) => {
+    res.send(users);
+  });
+
+  app.get('/users/:userId', (req, res) => {
+    const { userId } = req.params;
+    const foundUser = users.find(user => user.id === +userId);
```

Comment:
The find and filter functions are repeated several times (for users and expenses) in your code. It would be better to create generic functions and reuse them.